pkg/kn/flags/podspec_helper.go

```diff
@@ -865,3 +889,134 @@ func decodeContainersFromFile(filename string) (*corev1.PodSpec, error) {
+	parts := strings.Split(probeString, ";")
+	probe := &corev1.Probe{}
+	var err error
+	if len(parts) > 2 {
+		return nil, fmt.Errorf("unexpected probe format detected")
+	}
+	if len(parts) == 2 && len(parts[1]) > 0 {
+		probe, err = resolveProbeOptions(parts[1])
+		if err != nil {
+			return nil, err
+		}
+	}
+	handler, err := resolveProbeHandler(parts[0])
+	if err != nil {
+		return nil, err
+	}
+	probe.ProbeHandler = *handler
+
+	return probe, nil
+}
+
+// resolveProbeOptions parses probe commons options
+func resolveProbeOptions(probeString string) (*corev1.Probe, error) {
+	options := strings.Split(probeString, ",")
+	mappedOptions, err := util.MapFromArray(options, "=")
+	if err != nil {
+		return nil, err
+	}
+	probe := &corev1.Probe{}
+	for k, v := range mappedOptions {
+		// Trim & verify value is convertible to int
+		intValue, err := strconv.ParseInt(strings.TrimSpace(v), 0, 32)
+		if err != nil {
+			return nil, fmt.Errorf("not a nummeric value for parameter '%s'", k)
+		}
+		// Lower case param name mapping
+		switch strings.TrimSpace(strings.ToLower(k)) {
+		case "initialdelayseconds":
+			probe.InitialDelaySeconds = int32(intValue)
+		case "timeoutseconds":
+			probe.TimeoutSeconds = int32(intValue)
+		case "periodseconds":
+			probe.PeriodSeconds = int32(intValue)
+		case "successthreshold":
+			probe.SuccessThreshold = int32(intValue)
+		case "failurethreshold":
+			probe.FailureThreshold = int32(intValue)
+		default:
+			return nil, fmt.Errorf("not a valid probe parameter name '%s'", k)
+		}
+	}
+	return probe, nil
+}
+
+// resolveProbeHandler parses probe handler options
+func resolveProbeHandler(probeString string) (*corev1.ProbeHandler, error) {
+	if len(probeString) == 0 {
+		return nil, fmt.Errorf("no probe parameters detected")
+	}
+	probeParts := strings.Split(probeString, ":")
+	if len(probeParts) > 4 {
+		return nil, fmt.Errorf("too many probe parameters provided, please check the format")
+	}
+	var probeHandler *corev1.ProbeHandler
+	switch probeParts[0] {
+	case "http", "https":
+		if len(probeParts) != 4 {
+			return nil, fmt.Errorf("unexpected probe format, please use 'http:host:port:path'")
+		}
+		handler := corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{},
+		}
+		if probeParts[0] == "https" {
+			handler.HTTPGet.Scheme = v1.URISchemeHTTPS
+		}
+		handler.HTTPGet.Host = probeParts[1]
+		if probeParts[2] != "" {
+			// Cosmetic fix to have default 'port: 0' instead of empty string 'port: ""'
+			handler.HTTPGet.Port = intstr.Parse(probeParts[2])
+		}
+		handler.HTTPGet.Path = probeParts[3]
+
+		probeHandler = &handler
+	case "exec":
+		if len(probeParts) != 2 {
+			return nil, fmt.Errorf("unexpected probe format, please use 'exec:<exec_command>,<exec_command>'")
+		}
+		if len(probeParts[1]) == 0 {
+			return nil, fmt.Errorf("at least one command parameter is required for Exec probe")
+		}
+		handler := corev1.ProbeHandler{
+			Exec: &corev1.ExecAction{},
+		}
+		cmd := strings.Split(probeParts[1], ",")
```

Comment:
Should we allow for escaping `,` ?

Reply:
A corner case of "that should be fine..." statement? :)